File: vct-operation/platform_android/module/handlemodule/VCTHandleModule.java
package handlemodule;
import java.util.HashMap;
import android.util.Log;

import com.vincent.cocos2dx.VCTChannel;
import Manager.VCTModule;

public class VCTHandleModule extends VCTModule{
	
	private static VCTHandleModule mInstance = null;
	private final static String MODULE_NAME = "handlemodule";
	private HashMap<String, String> mCalbackMap = new HashMap<String, String>();
	
	public static VCTHandleModule getInstance(){
		if(mInstance == null){
			mInstance = new VCTHandleModule();
		}
		return mInstance;
	}
	
	private VCTHandleModule(){
		super(MODULE_NAME);
		this.RegisterMethod("register", "registerHandle");
		this.RegisterMethod("trigger", "triggerHandle");
	}
	
	public String registerHandle(final String param,final String callback){
		if(param.isEmpty()){
			Log.i("AndroidLog","[handlemodule] register must provide a param as a key");
			return "";
		}
		if(callback.isEmpty()){
			Log.i("AndroidLog","[handlemodule] register must provide a callback");
			return "";
		}
		mCalbackMap.put(param, callback);
		return "";
	}
	
	public String triggerHandle(final String param,final String callback){
		if(param.isEmpty()){
			Log.i("AndroidLog","[handlemodule] trigger must provide a param as a key");
			return "";
		}
		if(!mCalbackMap.containsKey(param)){
			Log.i("AndroidLog","[handlemodule] trigger " + param + " is not register");
			return "";
		}
		String registeredCallback =  mCalbackMap.get(param);
		mCalbackMap.remove(param);
		VCTChannel.Response("", registeredCallback);
		return "";
	}
}

File: platform_ios/module/handlemodule/VCTHandleModule.m
//
//  VCTHandleModule.m
//  demo
//
//  Created by Vincent on 15/7/20.
//
//

#import "VCTHandleModule.h"
#import "VCTManager.h"
#define MODULE_NAME @"handlemodule"

@interface VCTHandleModule()
@property(nonatomic,retain) NSMutableDictionary *callbackDic;
@end

@implementation VCTHandleModule
create_impl(VCTHandleModule)

- (void)dealloc {
    self.callbackDic = nil;
    [super dealloc];
}

- (id)init {
    if (self = [super initWithName:MODULE_NAME]) {
        self.callbackDic = [[NSMutableDictionary alloc] init];
        [self registerMethodWithName:@"register" Method:@selector(registerHandle:Callback:)];
        [self registerMethodWithName:@"trigger" Method:@selector(triggerHandle:Callback:)];
    }
    return self;
}

- (NSString *)registerHandle:(NSString *)param Callback:(NSString *)callback {
    if (param.length == 0) {
        NSLog(@"[handlemodule] register must provide a param as a key");
        return @"";
    }
    if (callback.length == 0) {
        NSLog(@"[handlemodule] register must provide a callback");
    }
    [self.callbackDic setObject:callback forKey:param];
    return @"";
}

- (NSString *)triggerHandle:(NSString *)param Callback:(NSString *)callback {
    if (param.length == 0) {
        NSLog(@"[handlemodule] trigger must provide a param as a key");
        return @"";
    }
    NSString *registeredCallback = [[self.callbackDic objectForKey:param] retain];
    if (nil == registeredCallback) {
        NSLog(@"[handlemodule] trigger %@ is not register",param);
    }else {
        [self.callbackDic removeObjectForKey:param];
        [[VCTManager instance] response:@"" Callback:registeredCallback];
    }
    return @"";
}
@end
